apps/openmw/mwscript/aiextensions.cpp

```diff
@@ -277,6 +279,7 @@ namespace MWScript
                     MWMechanics::Stat<int> stat = ptr.getClass().getCreatureStats(ptr).getAiSetting(mIndex);
                     stat.setModified(value, 0);
                     ptr.getClass().getCreatureStats(ptr).setAiSetting(mIndex, stat);
+                    ptr.getClass().setBaseAISetting(ptr.getCellRef().getRefId(), mIndex, value);
```

Comment:
Not sure if it works in this way in Morrowind. From what I can tell, setFight and so on can affect all aready spawned actors with given ID - if I remebered correctly, Morrowind just does not copy base AI stats from ESM record.

Reply:
According to rot SetFight and so on don't affect actors with the given ID that have already been loaded, they only affect the calling actor and actors that weren't loaded yet.

Reply:
Then just make sure that it works properly for guards when player refuses to surrender. (wjen `setfight 100` in dialogue makes all guards in sight hostile. So far we use a hack - if an NPC with "guard" class fights player, we force other guards attack player too, what has side effects.

Reply:
Yeah. Morrowind doesn't have that.

Reply:
Guard AI in vanilla is pretty inconsistent (in my experience anyway). I don't think it should be part of this PR though.

Reply:
Two man use-cases of SetFight on base actor records:
1. When you need to make every actor with given ID hostile (Morrowind.esm uses it for guards dialogues).
2. When you need to make every actor with given ID *non*-hostile (some mods use it to make for example, Dagoth creatures non-hostile when you join Sixth House).

These cases are the main reason to modify AI settings for base actors. They should work in the way that users and content creator would consider to be correct. I'd suggest to ask modmakers about how these cases are supposed to work, rather just claim them to be inconsistent and move along.

Reply:
All I'm saying is that guard AI is inconsistent and that it has nothing to do with mutating base records. Refusing to surrender does not `setfight 100` (it's commented out). Wearing Ordinator armor or having a bounty over 5000 does use `setfight` though.

From what little testing I've done, there appears to be no difference between implicit and explicit calls. (Doesn't mean there aren't any but... it's a positive sign.)